### ISSUE TITLE
H2A Megalo - Projectiles cannot be referenced

### ISSUE THREAD

**Issue by Gamergotten** [Joint-Issue-Tracker/Joint-Issue-Tracker#42]
### Issue type: Is this a bug or feature request?

Bug

### Which game's editing kit is affected by the issue?

H2AEK

### Which tools are affected by the issue?

- [ ] Guerilla (`guerilla.exe`)
- [ ] Foundation (`Foundation.exe`)
- [ ] Sapien (`sapien.exe`)
- [ ] Standalone (`halox_tag_test.exe`)
- [ ] Tool (`tool.exe`)
- [ ] Tool Fast (`tool_fast.exe` - unsupported)
- [X] MegaloEdit (`MegaloEdit.exe`)
- [ ] In-game (MCC)

### Issue description

in H2A, projectiles have no ability to interact with projectile teleporters nor is it possible to interact with them with megaloscript, because they are not properly identified as objects or something like that

in halo reach, they exist as objects that you can reference (so you can do things like attaching, spin around, push upwards etc)
and in reach, they also can go through teleporters (when specified with the projectile teleporter attribute)

i believe both issues are linked

### Reproduction steps

compile a gamemode that attempts to find projectiles, this can be done with code like this (code untested, it was a slight modification of code used in a gamemode to manipulate projectiles in reach)

```
; this label will access ALL objects, including projectiles
map_object everything 
end

trigger everything 
	temporary number health_check 0	
	temporary object this current_object 

	action object_get_health current_object health_check
	condition if health_check > 1 
	condition if this.is_projectile == 0
		action set health_check = 0
		; projectiles do not exist in the 'for_each object' trigger, so we can exclude this object if it is an object in that list
		action for_each object 
			condition if current_object == this		
				action set health_check = 1
		end
		condition if health_check == 0 
		; if true, this object is a projectile, run appropriate behavior, for this example, we will delete
		action object_destroy current_object
end
```

alternatively, load into forge and setup a projectile teleporter, attempt to throw grenades and/or projectiles through it

### How often does the issue occur?

100%

### Issue impact

not a major issue, but a lot of cool concepts do stem from the ability to find player's projectiles
mainly binding functionality to player's grenades
like spawning things where they throw their grenades

### Additional comments

this is not fully test for H4, but H4 suffers this issue with _some_ projectiles, like grenades, however others are fully able to go through projectile teleporters

**Comment by Gamergotten** [comment]
tested in H4, using a modified version of that script (health check removed, so we can access EVERY projectile, and not just ones with health)

some projectiles work, where others do not work, list of non working projectiles:
```
magnum
assault rifle
battle rifle
boltshot
charged boltshot
DMR
sniper
shotgun
carbine
lightrifle
beam rifle
binary rifle 
plasma grenade
frag grenade
pulse grenade
```

notably: the guns projectiles all expire instantly, so it is expected that you cant reference their projectiles
however the grenades do not expire, and you cannot reference them
and further, all projectiles that can be referenced can go through teleporters
and the ones that cant be referenced, cannot go through teleporters

TLDR: grenades cannot be referenced (by megalo) in H4, but every other projectile that exists for more than a tick do

mode used to test this
[!  proj testing.zip](https://github.com/Joint-Issue-Tracker/Joint-Issue-Tracker/files/9793866/proj.testing.zip)
